### PR TITLE
CI setup with integration tests

### PR DESCRIPTION
**Pull request by prometheansacrifice**

### CODE REVIEW

**Comment by avsm** [comment]
This lgtm!  We can investigate non-Travis options as a future change.

**Comment by imbsky** [comment]
GitHub Actions is a good choice. (But TravisCI is still the best choice when testing for many architectures. But something like GitHub Actions is sufficient for this project.)

**Comment by imbsky** [comment]
By the way, I don't want to use JavaScript for E2E tests.

**Comment by prometheansacrifice** [comment]
I'm open to suggestions. We have to keep in mind that e2e tests have to
spawn a vscode instance for them to be meaningful

**Comment by prometheansacrifice** [comment]
@avsm Can you please run the CI for me? I'm not sure I have access.

**Comment by rgrinberg** [comment]
@prometheansacrifice Would you mind resolving the conflicts?

I'll make sure the CI is on with Anil today.

**Comment by prometheansacrifice** [comment]
Sure, no problem.

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,6 @@
 node_modules
 out/
 .log/
+*~
+.vscode-test/
+_build/
```
Comment (imbsky) [inline .gitignore]: It's better to insert a final newline.

File: tests/fixtures/sample-opam/sample-opam.opam [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+maintainer: "foobar"
+authors: "foobar"
+homepage: "foo"
+description: "Description"
+synopsis: "Synopsis"
+bug-reports: "https//foo.bar"
+depends: [
+  "ocaml"             {>= "4.04.2" & < "4.10.0"}
+  "base"              {>= "v0.12" }
+]
```
Comment (imbsky) [inline tests/fixtures/sample-opam/sample-opam.opam]: Same here.

**Comment by imbsky** [comment]
I'm rewriting the tests in TypeScript and the work is almost complete.

**Comment by prometheansacrifice** [comment]
@imbsky Can you please hold the rewrite for a few hours? I'm drafting a
proposal that might affect your work.

**Comment by imbsky** [comment]
Sure! The basic code has not changed, so I will merge it with TypeScript branch as soon as you finish your work.

**Comment by prometheansacrifice** [comment]
@imbsky Did you get a chance to take a look at #16 ? (I tried tagging you)

**Comment by imbsky** [comment]
@prometheansacrifice I saw it, and I'm referring to it in Slack.

**Comment by rgrinberg** [comment]
Let's hold off on rewriting the tests to TS if it makes it harder to merge the rest of the work @prometheansacrifice has upcoming. The port to github actions can also wait until we have introduced all the tests into this repository.

As a bit of a meta point, shall we only have 1 person working on the CI for now? There's plenty of work to go around and there's no need to step on each other's toes. @imbsky could you wait until Manas merges his work before improving the CI? There's plenty of other things that need to work, so let me know if you need some suggestions.

**Comment by imbsky** [comment]
Of course, well, I tweaked it to run the test without rewriting, so we should be able to run it correctly just by replacing the location of the code. (The reason for rewriting some of the existing code is to make the test setup code more compatible with the VSCode official setup.)

**Comment by prometheansacrifice** [comment]
@imbsky Would you like to review the test setup at vscode-merlin before it gets merged. It runs a bunch of checks for all three kinds of projects types - opam/esy/npm. I'd say it's more comprehensive. These (and the ones at vscode-merlin) tests are setup as per sample-lsp example from the official documentation. It so happens, even if one wanted too, one cannot deviate much from the example setup. There isn't much scope for deviation. More info at this [issue](https://github.com/arrowresearch/vscode-merlin/issues/33)

**Comment by imbsky** [comment]
I also think it's good to setup according to the official example.

**Comment by rgrinberg** [comment]
@prometheansacrifice could you rebase?

**Comment by prometheansacrifice** [comment]
This PR can be closed since it's not as exhaustive as #19 Closing it

**Pull request closed without merging**